### DISCUSSION TITLE
How I can hide the "unpublish" button from a collection? 🙏🏻

### DISCUSSION THREAD

**Discussion by alexhito** [payloadcms/payload#5484] [Feature Requests & Ideas]
Hi, I have a collection that users can only create once and can no longer edit it, but I realized that an "unpublish" button appears and I must hide it so that they cannot enter and edit that document again. Could anyone help me please?
I have to hide or remove the "unpublish" text inside the document data too.
Note: I need the drafts enabled too.
![Screenshot 2024-03-26 at 19 07 33](https://github.com/payloadcms/payload/assets/57009193/7f39bb30-f623-4aae-99b0-5dfdcf890222)
![Screenshot 2024-03-26 at 19 07 50](https://github.com/payloadcms/payload/assets/57009193/db6a60ee-d672-4824-8303-b84873ed003b)

**Comment by DanRibbens** [comment]
I haven't tested this but it should be possible with `access.update` if you do something like this:

```
const Pages: CollectionConfig = {
  slug: 'pages',
  access: {
    update: ({ data, req }) => {
     // an admin user can do anything:
     if (req.user.isAdmin) return true

    // when the incoming data is status: draft, return a query to be sure that the document is not already published, a draft stays draft
      if (data._status === 'draft' ) {
        // this will not match any published docs and so the status cannot change.
        return { _published: { equals: 'draft' } }
      }

      // published docs can still be updated published
      return true
    },
  },
  fields: [
 // ... etc.
}
```

There more needed to disable the button in the UI, though I think this will prevent the action if they do try it. You might be able to conditionally hide the button in the UI, but I'd have to dig more to see if we have a selector that would work further too.

If you can't get it to work let me know and we can revist this as a new feature.

**Reply by michalcizek** [reply]
@DanRibbens Hey I have stumbled upon this, this works nicely, even takes care of the UI. The problem is, after someone publishes the document, non-admin user is not able to create a new draft version and keep the published version untouched. Is there a solution for that?
